### ISSUE TITLE
[BUG] Flyte array plugin fails with "ResourceDeletedExternally"

### ISSUE THREAD

**Issue by pablocasares** [flyteorg/flyte#5272]
### Describe the bug

In some workflows executions using the map_task the workflow fails because of
`
[498]: code:"ResourceDeletedExternally" message:"resource not found, name [ingestion-pipeline-production/jtep6yfsp4re7teyxnfs-n4-0-498]. reason: pods \"jtep6yfsp4re7teyxnfs-n4-0-498\" not found"
[500]: code:"ResourceDeletedExternally" message:"resource not found, name [ingestion-pipeline-production/jtep6yfsp4re7teyxnfs-n4-0-500]. reason: pods \"jtep6yfsp4re7teyxnfs-n4-0-500\" not found"
[501]: code:"ResourceDeletedExternally" message:"resource not found, name [ingestion-pipeline-production/jtep6yfsp4re7teyxnfs-n4-0-501]. reason: pods \"jtep6yfsp4re7teyxnfs-n4-0-501\" not found"
[502]: code:"ResourceDeletedExternally" message:"resource not found, name [ingestion-pipeline-production/jtep6yfsp4re7teyxnfs-n4-0-502]. reason: pods \"jtep6yfsp4re7teyxnfs-n4-0-502\" not found"
[503]: code:"ResourceDeletedExternally" message:"resource not found, name [ingestion-pipeline-production/jtep6yfsp4re7teyxnfs-n4-0-503]. reason: pods \"jtep6yfsp4re7teyxnfs-n4-0-503\" not found"
... and many more.
`

This seems to be happening because the pod is being removed before the final status is checked by the propeller.
Those pods mentioned didn't failed. They were executed normally but it seems they are being cleaned up before the propeller reads the Succeeded final status.

### Expected behavior

Flyte array plugin should not fail because of this. The pod final status should be checked before being cleaned.

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

**Comment by andresgomezfrr** [comment]
> this suggests that k8s is under very heavy load and is not respecting the finalizers

Is it possible? I think that even with a very heavy load k8s should respect the finalizers.

> Do you have a notion of the order of concurrent Pod executions when this behavior occurs? It would explain why reproducing this is difficult.

thousands of pods more or less

**Comment by hamersaw** [comment]
> Is it possible? I think that even with a very heavy load k8s should respect the finalizers.
Yes, we see it relatively frequently with a massive number of Pods in a k8s cluster. My intuition is that is exactly what is happening here. A potential solution is to configure Flyte to execute over multiple k8s clusters to reduce the number of Pods per cluster.
